site/content/docs/main/troubleshooting.md

```diff
@@ -46,6 +46,8 @@ kubectl edit deployment/velero -n velero
 ...
 ```
 
+**Note:** Velero plugins run as binaries, i.e. after they are executed, either successful or not, they exit. So, if you see **received EOF, stopping recv loop** messages in debug logs, that does not mean an error occurred, just that a plugin finished executing.
```

Comment:
Are we really sure that this error is benign? My understanding is that Velero server process starts each plugin as a separate process when it comes up and the plug-in processes keep running. If they exit, it must be due to some problem. But it is possible I am missing something here and it will be great if some one clarifies the exact lifecycle of plugin processes. 

BTW, "plugins run as separate processes" is preferable to "run as binaries".

Reply:
I will make that change in the next commit, thanks

As an example, every minute Velero checks if Backup Storage Location is available (marking as valid, invalid or unknown). Each time it finishes the check,  you are able to see this message (if log level is debug).

Reply:
Interesting. @sseago @shubham-pampattiwar Can you please shed some light on how Velero starts plug-ins? I think it uses https://github.com/hashicorp/go-plugin and I thought that the plug-in gRPC server will be up and running all the the time just like Velero server. But it doesn't seem to be the case?

Reply:
You are correct @draghuram, the exactly line the message is triggered https://github.com/hashicorp/go-plugin/blob/c57883964917bfb33721f5c5c5c8eeb98fe933e7/grpc_stdio.go#L142

Reply:
@draghuram In my experience, when turning Debug on, the plugin processes exit after a reconcile. So for BSL validation, it exits after checking. For running a backup, all of the plugin invocations for a given BackupItemAction plugin are invoked on the same plugin process, but once backup processing completes, the plugin exits.

Reply:
Got it, thanks.